H_ecto_models.md

```diff
@@ -597,6 +597,62 @@ end
 ```
 Because we declared a relationship in `HelloPhoenix.User`, `%User{}` will now contain a videos property which starts out as an unloaded relationship. In order to properly display it in `render`, we'll need to tell Ecto to preload the field. Note that `Repo.preload/2` is smart enough to work on just one model or collection of them.
 
+### Working with Callback Hooks
+
+Continuing with our video sharing app example, suppose our service takes off and we now have a healthy community of viewers, content-creators, and a small team of volunteer moderators who review new uploads and approve the good ones while keeping out the illegal ones. But certain malicious users have figured out a way around our system by first uploading a "safe" video, waiting for our moderators to approve it, then changing the video to something ilegal.
```

Comment:
I like the idea of using the approved_at field!

I think we can express this idea more clearly with fewer words. Here's a quick first pass at the first two paragraphs. What do you think?

Many web applications require moderation for user submitted content. Our `Video` model has an `approved_at` field to signify that a moderator has approved the video. The question is, what happens if the user edits the video? Ideally, we'd like to be able to "null out" a video's `approved_at` field when this happens. Fortunately, Ecto provides us with a handful of life-cycle callbacks to achieve this.